### PR TITLE
168 improvments

### PR DESCRIPTION
**Pull request by kmarszalek**

### CODE REVIEW

File: s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java [hunk 1]
```diff
@@ -143,8 +151,23 @@
...
     @PostMapping("/institutions/{institution}/users")
-    public ResponseEntity<?> addUserToInstitution(@RequestBody @Valid CreateUserWithGroupsRequest request) {
-
+    public ResponseEntity<?> addUserToInstitution(@RequestBody @Valid CreateUserWithGroupsRequest request,
+                                                  @PathVariable("institution") String institutionSlug) {
+        AppUser appUser = AppUser.builder()
+                .name(request.getName())
+                .surname(request.getSurname())
+                .email(request.getEmail())
+                .password(passwordEncoder.encode("passTemporary"))
```
Comment (jswk) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java]: Maybe better, if it was random. Unless it is only for the demo?

File: s4e-backend/src/main/java/pl/cyfronet/s4e/controller/request/CreateUserWithGroupsRequest.java [hunk 2]
```diff
@@ -5,11 +5,18 @@
...
 public class CreateUserWithGroupsRequest {
     @Email
     @NotEmpty
     private String email;
+    @NotEmpty
+    private String name;
+    @NotEmpty
+    private String surname;
+
+    private Set<String> slugsGroups;
```
Comment (jswk) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/controller/request/CreateUserWithGroupsRequest.java]: -> groupSlugs?

File: s4e-backend/src/main/java/pl/cyfronet/s4e/service/AppUserService.java [hunk 3]
```diff
@@ -39,4 +41,15 @@ public AppUser update(AppUser appUser) {
     public Optional<AppUser> findById(Long id) {
         return appUserRepository.findById(id);
     }
+
+    @Transactional
+    public void saveWithGroupUpdate(AppUser appUser) {
+        appUserRepository.save(appUser);
```
Comment (jswk) [inline s4e-backend/src/main/java/pl/cyfronet/s4e/service/AppUserService.java]: You don't catch DataIntegrityViolationException here, as in the standard method ::save?

File: s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java (hunk 1 above)
Comment (kmarszalek) [reply s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java]: only for demo

Comment (jswk) [reply s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java]: Add a FIXME there saying to fix it later

File: s4e-backend/src/main/java/pl/cyfronet/s4e/controller/request/CreateUserWithGroupsRequest.java (hunk 2 above)
Comment (kmarszalek) [reply s4e-backend/src/main/java/pl/cyfronet/s4e/controller/request/CreateUserWithGroupsRequest.java]: and not groupsSlugs?

Comment (jswk) [reply s4e-backend/src/main/java/pl/cyfronet/s4e/controller/request/CreateUserWithGroupsRequest.java]: I'd stick to my version, you only use the plural form on the last component usually

**Pull request merged**